ipa-core/src/protocol/ipa_prf/aggregation/mod.rs

```diff
@@ -181,21 +181,25 @@ where
 
     let aggregation_input = Box::pin(
         row_contribution_chunk_stream
-            // The final chunk from the previous stage is padded with zero-credit records. Rather
-            // than transpose out of vectorized form, flatten the chunked stream, discard the
-            // trailing records, and transpose again for final aggregation, we instead call into_raw
-            // to get the padded record chunk and transpose directly to the form we need for the
-            // final stage. Including the zero-credit padding records does not affect the final
-            // output.
-            .then(|fut| async move { fut.await.map(Chunk::into_raw) })
-            .map_ok(|chunk| {
-                // This is the aggregation intermediate transpose, see the function comment.
-                Vec::transposed_from(chunk.as_slice()).unwrap_infallible()
+            // Rather than transpose out of record-vectorized form and then transpose again back
+            // into bucket-vectorized form, we use a special transpose (the "aggregation
+            // intermediate transpose") that combines the two steps.
+            //
+            // Since the bucket-vectorized representation is separable by records, we do the
+            // transpose within the `Chunk` wrapper using `Chunk::map`, and then invoke
+            // `Chunk::into_iter` via `try_flatten_iters` to produce an unchunked stream of
+            // records, vectorized by buckets.
+            .then(|fut| {
+                fut.map(|res| {
+                    res.map(|chunk| {
+                        chunk.map(|data| Vec::transposed_from(data.as_slice()).unwrap_infallible())
+                    })
+                })
             })
```

Comment:
Even with the above comment, I'm sure I couldn't tell you the type of `fut`, `res`, `chunk` or `data`. I'm not sure how to fix this. Maybe a comment for each saying what the structure of that data object is?